### DISCUSSION TITLE
Should we increase the hardcoded limit of 3 models in embeddings query from 3 to a higher number?

### DISCUSSION THREAD

**Discussion by codeblech** [pragunbhutani/dbt-llm-agent#36] [Ideas]
`/dbt_llm_tools/vector_store.py`
```
def query_collection(
        self, query: str, n_results: int = 3
    ) -> list[ParsedSearchResult]:
```

**Comment by SohamKukreti** [comment]
n_results in this function doesn't seem to have an effect on the number of models fetched at least when i experimented.
```
def query_collection(
        self, query: str, n_results: int = 3
    ) -> list[ParsedSearchResult]:
```
I asked the chatbot `Which part of the business generates the most revenue?`
it gave the following answer:
```
SELECT
    session_channel_grouping,
    SUM(total_transaction_value) AS total_revenue
FROM
    fct_channel_performance
GROUP BY
    session_channel_grouping
ORDER BY
    total_revenue DESC
LIMIT 1;
```
And in the background query_collection fetched 4 closest models with ids:
```
dim_customer
fct_channel_performance
fct_sessions
dim_customer_identity
```

**Reply by pragunbhutani** [reply]
Can you check if the Param is being forwarded to the collection.query method? It's possible that I may have hardcoded the value of 4 there.

As for the ideal solution, I think we should have the vector store return all the values where the cosine distance is above an acceptable threshold value. We can determine this value experimentally.

**Reply by codeblech** [reply]
the parameter n_result is forwarded...
```
def query_collection(
        self, query: str, n_results: int = 3
    ) -> list[ParsedSearchResult]:
    ...
    search_results = self.__collection.query(
        query_texts=[query],
        n_results=n_results,
        include=["documents", "distances", "metadatas"],
        )
    ...
```

**Reply by codeblech** [reply]
found it! you're right... you've indeed hardcoded n_results=4 in `/client/pages/chat.py`
```
def get_matching_models(query):
    return vector_store.query_collection(query=query, n_results=4)
```

**Reply by codeblech** [reply]
> As for the ideal solution, I think we should have the vector store return all the values where the cosine distance is above an acceptable threshold value. We can determine this value experimentally.  

yes, this approach seems better

**Reply by SohamKukreti** [reply]
I'll create an issue for this to be tracked and worked upon in the future, is that fine?

**Reply by codeblech** [reply]
yep

**Comment by pragunbhutani** [comment]
This is no longer relevant because the agentic workflow can fetch more models when needed.
